packages/aws-cdk-lib/aws-sns/lib/topic.ts

```diff
@@ -217,6 +227,12 @@ export class Topic extends TopicBase {
       cfnTopicName = this.physicalName;
     }
 
+    if (props.signatureVersion !== undefined) {
+      if (props.signatureVersion !== '1' && props.signatureVersion !== '2') {
```

Comment:
```suggestion
    if (
      props.signatureVersion &&
      !Token.isUnresolved(props.signatureVersion) && 
      props.signatureVersion !== '1' &&
      props.signatureVersion !== '2'
    ) {
```
Let's also verify that the variable has a resolved value.